src/peft/utils/hotswap.py

```diff
@@ -87,14 +368,48 @@ def hotswap_adapter_from_state_dict(model, state_dict, adapter_name, parameter_p
 
     # actual swapping
     for key, new_val in state_dict.items():
+        # get LoRA parent module name by removing the 'lora_*.<adapter-name>.weight' part
+        module_name = ".".join(key.split(".")[:-3])
+        module = model.get_submodule(module_name)
+
+        # swap alpha/scaling
+        r_key = get_pattern_key(config.rank_pattern.keys(), key)
+        alpha_key = get_pattern_key(config.alpha_pattern.keys(), key)
+        rank = config.rank_pattern.get(r_key, config.r)
+        alpha = config.alpha_pattern.get(alpha_key, config.lora_alpha)
+        if config.use_rslora:
+            scaling = alpha / math.sqrt(rank)
+        else:
+            scaling = alpha / rank
+        _update_scaling(module, adapter_name=adapter_name, scaling=scaling)
+
+        # swap actual weights
         # no need to account for potential _orig_mod in key here, as torch handles that
         old_val = attrgetter(key)(model)
-        if is_compiled:
-            # Compiled models don't work with swap_tensors because there are weakrefs for the tensor. It is unclear if
-            # this workaround could not cause trouble but the tests indicate that it works.
+        if not is_compiled:
+            torch.utils.swap_tensors(old_val, new_val)
+            continue
+
+        # Compiled models don't work with swap_tensors because there are weakrefs for the tensor. It is unclear if
+        # this workaround could not cause trouble but the tests indicate that it works.
+        if old_val.shape == new_val.shape:
             old_val.data = new_val.data
         else:
-            torch.utils.swap_tensors(old_val, new_val)
+            if old_val.dim() != 2:
+                # TODO conv2d
+                raise NotImplementedError
+            if old_val.shape[0] > new_val.shape[0]:
+                old_val.data.fill_(0)
```

Comment:
(nit): Would it make sense to have `0` assigned as a `PAD_VALUE` var and use it?